controllers/pattern_controller.go

```diff
@@ -505,3 +535,73 @@ func (r *PatternReconciler) actionPerformed(p *api.Pattern, reason string, err e
 	}
 	return r.onReconcileErrorWithRequeue(p, reason, err, nil)
 }
+
+// MBP-184 Implementation
+// applyPatternAppDetails retrieves the status for the Pattern applications created in ArgoCD.
+// Once the Application Status is retrieve it appends the information
+// to the array and the CR is updated.
+func (r *PatternReconciler) applyPatternAppDetails(client argoclient.Interface, input *api.Pattern) (*api.Pattern, error) {
+	output := input.DeepCopy()
+	var applicationInfo api.PatternApplicationInfo
+
+	// Buid the namespace from the Pattern ObjectMeta.Name and Spec.ClusterGroupName
+	ns := output.ObjectMeta.Name + "-" + output.Spec.ClusterGroupName
```

Comment:
What if instead we look for applications in any namespace with a label based on the pattern name?

Eg. add `pattern: {{ $.Values.global.pattern }}` as a label to
https://github.com/validatedpatterns/common/blob/main/clustergroup/templates/plumbing/applications.yaml#L120

Might be less fragile that keeping this logic in sync.